bftengine/src/bcstatetransfer/RangeValidationTree.cpp

```diff
@@ -0,0 +1,919 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the
+// LICENSE file.
+
+#include <queue>
+#include <algorithm>
+
+#include "RangeValidationTree.hpp"
+#include "STDigest.hpp"
+
+using namespace std;
+using namespace concord::serialize;
+
+// Each node, even 0 level nodes, holds a current value and an initial value.
```

Comment:
I suggest to move this into the header, there is explenation there, why having 2 places with an overview?
combine the both together.